### PR TITLE
bugfix: improve the assignment of conditional variable values

### PR DESCRIPTION
**Pull request by msmikesm**

### CODE REVIEW

**Comment by NicolasCARPi** [comment]
Hello, it seems your devtool change makes the check fail.

File: test/writer_test.js [hunk 1]
```diff
@@ -30,4 +30,24 @@ describe('test writer module', function () {
     const writer = new Writer({}, app);
     return writer.write();
   });
+
+  it('getIndexContent shouldn\'t throws error', async function () {
```
Comment (NicolasCARPi) [inline test/writer_test.js:34]: I don't understand this test. Shouldn't it be "getIndexContent() should throw an error if .... " in the description?

File: test/writer_test.js [hunk 2]
```diff
@@ -30,4 +30,24 @@ describe('test writer module', function () {
...
+    const app = {
+      log: logger,
+    };
+    const writer = new Writer({}, app);
+    assert.throws(writer.getIndexContent, Error);
+  });
+
+  it('should generate inline source maps if debug option is true', async function () {
```
Comment (NicolasCARPi) [inline test/writer_test.js:42]: this test is missing an assertion

Comment (msmikesm) [reply test/writer_test.js:42]: oh, I'm sorry I'll fix it

**Review by msmikesm — COMMENTED** [review]
Hey @NicolasCARPi , 
I made the corrections.
Regarding the test without assertion please look at your first test, it doesn't have it either, it only checks the correct execution of the program by running it.

**Comment by NicolasCARPi** [comment]
> Regarding the test without assertion please look at your first test, it doesn't have it either, it only checks the correct execution of the program by running it.

Yeah, tests are not really good, but it doesn't mean that we should not thrive for improvements for new code. Still, I agree that if it "compiles" ok, then it's mostly good enough.

**Comment by NicolasCARPi** [comment]
Thanks!

**Pull request merged**